hikari/events/channel_events.py

```diff
@@ -137,18 +137,34 @@ def guild_id(self) -> snowflakes.Snowflake:
         """
 
     @property
-    def guild(self) -> typing.Optional[guilds.GatewayGuild]:
-        """Get the cached guild that this event relates to, if known.
+    def available_guild(self) -> typing.Optional[guilds.GatewayGuild]:
```

Comment:
I'm not sure I like this, I think `guild` is fine here.

I feel like this will possibly get confusing for people, as it forces their code to make assumptions about the gateway guild state (which may or may not correlate to the state on the REST API).

I would hope you can usually assume that if you got a channel event, it would still be available. But this kind of obfuscates how you access a guild here.

I think it would be safer to just do
```py
@property
def guild() -> typing.Optional[guilds.GatewayGuild]:
    return self.app.cache.get_available_guild(self.guild_id) or self.app.cache.get_unavailable_guild(self.guild_id)
```